sql/src/main/java/org/apache/druid/sql/calcite/view/DruidViewMacro.java

```diff
@@ -58,6 +58,7 @@ public TranslatableTable apply(final List<Object> arguments)
   {
     final RelDataType rowType;
     try (final DruidPlanner planner = plannerFactory.createPlanner(viewSql, new QueryContext())) {
+      planner.validate(false);
```

Comment:
I didn't realize that view expansion was done with an empty query context. I suppose this makes sense: it ensures the view is the same for everyone.

IMO, it's better to set this to always `true` instead of always `false`. Right now, it doesn't matter either way, because the context is empty. But if it's ever made nonempty, then `true` is a safer constant value because it defaults to the more restrictive security mode. If that doesn't make sense in the future, then the future person that is adding the context parameters can sort out what's best.

A comment would be helpful too, like:

```
// Since the context is empty, it doesn't matter if the authorizeContextParams flag is true or false.
```

Reply:
>As part of the auth cleanup, I moved the authorizeContextParams flag to the authorization step, and clearly stated that a statement (or view) can be prepared without authorization. Only execution (AKA "plan") needs authorization.

Hmm, currently `DruidStatement` calls `validateAndAuthorize` for both `prepare` and `execute`, are you saying that `prepare` no longer authorizes or that it never did? (because looking at the current code it seems to do 

>The standard way to handle views is to assign them an owner. Views run with the owner's permissions. Queries that use the view must be authorized against the view, not against the resources which the view uses. Of course, Druid has no concept of users, so the idea of "owner" is ill-defined. Maybe we stash a bundle of permissions with the view or some such? That's a question for another time.

There is a 'VIEW' resource that views are authorized against, not the resource that the view query uses, which i think is more or less the same thing?

Reply:
What I saw in the code is that the Avatica path did validate, authorize, prepare. But, the view path did validate, prepare but no authorize (it has no auth result or request). The flow before and after the change is the same. For Avatica, the  path is validate, authorize, prepare. The query path is validate, authorized, plan. The view path is validate, prepare (with no authorize). Does this sound right?

Reply:
Oh, I misunderstood what you were saying. Yes within the view's usage of the planner it is basically unauthorized on purpose, because the view itself as a whole is authorized as a `ResourceType.VIEW` by the `SqlLifecycle` of the query using the view. The planner usage here is to get the row type information so the query planner using the view can do its thing. #10812 has details on view authorization stuff (which is also the PR that left the comments that this PR is all about).